packages/flutter/lib/src/animation/tween.dart

```diff
@@ -256,6 +256,35 @@ class Tween<T extends dynamic> extends Animatable<T> {
   T lerp(double t) {
     assert(begin != null);
     assert(end != null);
+    assert(() {
+      // Assertions that attempt to catch common cases of tweening types
+      // that do not conform to the Tween requirements.
+      dynamic? result;
+      try {
+        result = begin + (end - begin) * t;
+        result as T;
+        return true;
+      } on NoSuchMethodError {
+        throw FlutterError.fromParts(<DiagnosticsNode>[
+          ErrorSummary('Cannot tween between $begin and $end.'),
+          ErrorDescription(
+            'The type ${begin.runtimeType} does not fully implement `+`, `-`, and/or `*`.',
```

Comment:
does not -> may not, since it's possible that it has a buggy implementation of `+` or something